src/frontend/hooks/useGetQuest.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery, useQueryClient } from 'react-query'
+
+export default function useGetQuest(questId: number | null) {
+  const queryClient = useQueryClient()
+  const queryKey = `getQuest:${questId}`
+  const query = useQuery(
+    queryKey,
+    async () => {
+      if (questId === null) {
+        return {}
+      }
+      const response = await window.api.getQuest(questId)
+      if (!response) return null
+      return response
+    },
+    {
+      refetchOnWindowFocus: false
+    }
```

Comment:
You might want to make this a dependent query: https://tanstack.com/query/latest/docs/framework/react/guides/dependent-queries, and only enable it if `questId` is defined.

Reply:
made the change! didn't know about this feature